app/src/main/java/com/android/partagix/model/notification/FirebaseMessagingService.kt

```diff
@@ -0,0 +1,337 @@
+import com.google.firebase.messaging.RemoteMessage
+import com.google.firebase.messaging.remoteMessage
+import java.io.IOException
+import java.sql.Date
+import okhttp3.Call
+import okhttp3.Callback
+import okhttp3.MediaType.Companion.toMediaTypeOrNull
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.RequestBody.Companion.toRequestBody
+import okhttp3.Response
+import org.json.JSONObject
+
+class FirebaseMessagingService(private val db: Database = Database()) : FirebaseMessagingService() {
+  private lateinit var requestPermissionLauncher: ActivityResultLauncher<String>
+  private var context = MainActivity.getContext()
+  private var navigationActions = App.getNavigationActions()
+
+  init {
+    Log.d(TAG, "context: $context")
+  }
+
+  /**
+   * Initializes the permissions (notifications) for the app.
+   *
+   * This function should be called during the MainActivity's onCreate() method. It asks the user
+   * for permission to show notifications and creates the notification channels, as described in
+   * `createChannels()`.
+   */
+  fun initPermissions() {
+    if (context == null) {
+      Log.e(TAG, "Context is not set, cannot initialize permissions")
+      return
+    }
+
+    requestPermissionLauncher =
+        context!!.registerForActivityResult(
+            ActivityResultContracts.RequestPermission(),
+        ) { isGranted: Boolean ->
+          if (isGranted) {
+            Log.d(TAG, "Permission granted")
+          } else {
+            // TODO: Inform user that that your app will not show notifications.
+            Log.d(TAG, "Permission denied")
+          }
+        }
+
+    askNotificationPermission()
+    createChannels()
+  }
+
+  /**
+   * Called when a message (a notification) is received.
+   *
+   * When the application is on the foreground, the notification is displayed as an alert, using the
+   * `notificationAlert()` function.
+   *
+   * @param remoteMessage Object representing the message received from Firebase Cloud Messaging.
+   */
+  override fun onMessageReceived(remoteMessage: RemoteMessage) {
+    super.onMessageReceived(remoteMessage)
+
+    if (context == null || navigationActions == null) {
+      Log.e(TAG, "Context or navigationActions is not set, cannot show notification")
+      return
+    }
+
+    val data = remoteMessage.data
+    val notificationBody = remoteMessage.notification
+
+    Log.d(TAG, "From: ${remoteMessage.from}, data: $data, notification: $notificationBody")
+
+    if (data.isNotEmpty() && notificationBody != null) {
+      val date =
+          try {
+            Date.valueOf(data["creationDate"])
+          } catch (e: IllegalArgumentException) {
+            Date(System.currentTimeMillis())
+          }
+
+      val notification =
+          Notification(
+              title = notificationBody.title ?: "",
+              message = notificationBody.body ?: "",
+              type = Notification.Type.NEW_INCOMING_REQUEST,
```

Comment:
Here should it always be NEW_INCOMING_REQUEST

Reply:
You are right, the type should not be hardcoded, but collected from the notification content itself. This is fixed now